### PR TITLE
Update docs for Strategy, Accelerator, Plugins

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

- Updates several outdated parts in the docs under the chapters Accelerator, Plugins, and Strategy.
- Updated graphics, after accelerator and precision was moved into the Strategy
- Updated references and links

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/extensions/accelerator.rst [hunk 1]
```diff
@@ -1,51 +1,111 @@
...
 
-.. warning:: The Accelerator API is in beta and subject to change.
-    For help setting up custom plugins/accelerators, please reach out to us at **support@pytorchlightning.ai**
 
 ----------
 
+Registering Accelerators
+------------------------
+
+If you wish to switch to a custom accelerator from the CLI without code changes, you can use the :class:`~pytorch_lightning.accelerators.registry.AcceleratorRegistry` to register your new accelerator under a shorthand name like so:
```
Comment (four4fish) [inline docs/source/extensions/accelerator.rst:86]: User could include `@classmethod def register_accelerators` in the customized accelerator class to register their accelerator too. I think suggesting user add register function in the customized class, and provide this as additional way is more clear?

Are we gonna update this one as well? https://pytorch-lightning.readthedocs.io/en/latest/advanced/strategy_registry.html

File: docs/source/extensions/accelerator.rst [hunk 2]
```diff
@@ -1,51 +1,111 @@
...
+
+.. code-block:: python
+
+    from pytorch_lightning.accelerators import AcceleratorRegistry
+
+    AcceleratorRegistry.register("xpu", XPUAccelerator, description="XPU Accelerator")
```
Comment (kaushikb11) [inline docs/source/extensions/accelerator.rst:92]: `XPUAccelerator` would need to implement the `register_accelerators` method

File: docs/source/extensions/plugins.rst [hunk 3]
```diff
@@ -18,6 +18,17 @@ There are three types of Plugins in Lightning with different responsibilities:
 - CheckpointIO Plugins
 - Cluster Environments
 
+You can make the Trainer use one or multiple plugins by adding it to the ``plugins`` argument like so:
```
Comment (four4fish) [inline docs/source/extensions/plugins.rst:21]: In this plugin session, should we mention that if no plugin is passed in, we will choose for user?

File: docs/source/extensions/strategy.rst [hunk 4]
```diff
@@ -16,20 +16,36 @@ The Strategy in PyTorch Lightning handles the following responsibilities:
 * Handles/owns optimizers and schedulers.
```
Comment (four4fish) [inline docs/source/extensions/strategy.rst:16]: 1. nit: L13 and L14 could be merged? 
2. for 16, Strategy also handles move model/data to device, model wrapping, forward, backwards, and optimizer_steps, will only mention `handles/owns optimizer/scheduler`  a bit confusing for users who doesn't know the lightning code base?

**Review by four4fish — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

File: docs/source/extensions/strategy.rst [hunk 5]
```diff
@@ -47,14 +63,74 @@ Training Strategies with Various Configs
     trainer = Trainer(accelerator="ipu", devices=8)
 
 
+The below table lists all relevant strategies available in Lightning with their corresponding short-hand name:
+
+
+.. list-table:: Strategy Classes and Nicknames
```
[the remark this answers is not in the document]
Comment (awaelchli) [reply docs/source/extensions/strategy.rst:69]: I don't think so, because these are not unique strategies but rather variations of options. However, I do think we can mention them in the section for the strategy registry.

File: docs/source/extensions/accelerator.rst (hunk 1 above)
Comment (awaelchli) [reply docs/source/extensions/accelerator.rst:86]: Yes, we may want to update the strategy registry too. Leaving it for a separate PR because it's a separate page atm.

File: docs/source/extensions/strategy.rst (hunk 4 above)
Comment (awaelchli) [reply docs/source/extensions/strategy.rst:16]: Going to leave it like this for now. Yes, we can list many more things but these are the core responsibilities and for now we can leave the intro brief.

> a bit confusing for users who doesn't know the lightning code base?

The strategy and accelerators API is more for advanced users and we assume they are not afraid to look at the internals.
